filebeat/input/filestream/prospector.go

```diff
@@ -82,6 +83,8 @@ func (p *fileProspector) Init(cleaner loginp.ProspectorCleaner) error {
 		return "", fm
 	})
 
+	p.ignoreInactiveSince = ignoreSince
```

Comment:
If possible we should reduce the need to modify the type on Init. Can the `Configure` function extract the new setting instead of the InputManager? In the end, this setting is a prospector setting only.

For reusability (if that is the motivation), we should switch to a coding pattern that allows us to compose/embedd functionality instead of building 'common' features around.

Reply:
The setting can be extracted by configure. But in this case, I also need to pass the timestamps (start time and first start time) to the Prospector to select the one configured. Passing down `beat.Info` to the `Prospector` feels too much and requires too big changes: passing `beat.Info` to `configure`, changing the signature of a clear function.

> For reusability (if that is the motivation), we should switch to a coding pattern that allows us to compose/embedd functionality instead of building 'common' features around.

Reusability is also part of my decision. I agree with you, when we are introducing new prospectors, we must revisit the `Init` function because it is going to be the same for the new ones, so it does not make sense to copy-paste the same code for all different `Prospectors`.

Reply:
In the `v2` input API we prefer to build a 'context' and pass it to run. Other subsystems are encouraged to define a custom context type. The context/environment is part of the interface, that allows us to push additional details.

Apropos. Check out [`(v2.Context).Agent`](https://github.com/elastic/beats/blob/master/filebeat/input/v2/input.go#L89). Is this what you need in the prospector run?

Reply:
Ha, that's exactly what I need! Thanks.